src/workerd/api/actor-state.c++

```diff
@@ -919,16 +919,20 @@ v8::Local<v8::Value> deserializeV8Value(
     // counting it against our internal storage error metrics but also throw a KJ exception rather
     // than a jsExceptionThrown error to avoid confusing the normal termination handling code.
     // We don't expect users to ever actually see this error.
-    JSG_REQUIRE(!tryCatch.HasTerminated(),
-        Error, "isolate terminated while deserializing value from Durable Object storage; "
-        "contact us if you're wondering why you're seeing this");
+    auto jsException = tryCatch.Exception();
+    if (tryCatch.HasTerminated()) {
+      KJ_LOG(ERROR, "actor storage deserialization failed", errorMsg, jsException,
+          key, buf.size(), buf.slice(0, std::min(static_cast<size_t>(3), buf.size())));
+      JSG_FAIL_REQUIRE(Error,
+          "isolate terminated while deserializing value from Durable Object storage; "
+          "contact us if you're wondering why you're seeing this");
```

Comment:
"Failed to deserialize value into isolate", we were hoping to make this message less misleading!